### DISCUSSION TITLE
issue: Warning: Can't perform a React state update on an unmounted component. (when trigger is called)

### DISCUSSION THREAD

**Discussion by maxy4u** [react-hook-form/react-hook-form#7474] [Q&A]
### Version Number

7.22.5

### Codesandbox/Expo snack

https://codesandbox.io/s/usefieldarray-virtual-input-forked-61i7e

### Steps to reproduce

1. Go to https://codesandbox.io/s/usefieldarray-virtual-input-forked-61i7e?file
2. Click on checkbox
3. click on append
4. Type something in input and click on x button to clear text of input. this will trigger validation.
5. There is a warning thrown in console upon trigger.
6. Comment the trigger call inside onClear.
7. Repeat step 1 to 5.

### Expected behaviour

Warning should not appear.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
update will `unmount` and `remount` with updated input, `trigger` need to used at `useEffect`

**Reply by maxy4u** [reply]
hi @bluebill1049  Sorry i didnt get above response of yours. Are you suggesting me to do these changes in my code?

**Reply by bluebill1049** [reply]
correct, basically field array will get unmounted and remounted during field array actions. It's safer to do at `useEffect` with `trigger`.

**Comment by maxy4u** [comment]
But the caveat is we need to trigger validation when user clear on clear icon which will clear the input value to "". So we decided to initiate trigger in out onClear/onClick handler.

Going by your approach we may have to write a useEffect whose dependency would be the value of this field. any change in the same would require the trigger to be called in this useEffect ?

Moreover, Are we using memo(fieldArray) ?

**Reply by maxy4u** [reply]
@bluebill1049  i have updated my sandbox please have a look and kindly weigh in your opinion.
https://codesandbox.io/s/usefieldarray-virtual-input-forked-61i7e?file=/src/App.js

Possible solution i figured
One is useEffect listening to see is firstName of firstField gets cleared. (but if we have multiple arrays then it is a challenge)
Another solution i though would defy warning would  be executing trigger asynchronously through setTimeout. again this is also not the ideal solution i would opt for.

Is there any other way? Like were handling some error like this in our app by creating a custom hook called useIsMounted() which will update a state only when component is mounted. Do you think we can do that in your code?

**Reply by bluebill1049** [reply]
hmmm to be honest this warning is not relevant as there is no memory leak, I think react is going to fix this warning which was only meant for something like an event subscription not getting unsubscribed. I did consider doing validation inside the lib when action is `trigger` at `useFieldArray`. I was thinking of something like: `append({}, { shouldValidate: true})` UX concerns caught me tho, it will validate the entire field array and impact to those fields which are yet interacted with. Let me try to figure out something... no guarantee it will be included in the lib tho.

**Reply by bluebill1049** [reply]
found a solution will be patched in the next release.

**Reply by maxy4u** [reply]
What is the solution just curious :) ?

**Reply by bluebill1049** [reply]
eat this in the library :( increase bundlesize just so we don't see this warning, i hope `react` team will improve/remove this warning soon, as this is not really a memory leak. it's just a function call..
